pkg/cmd/join/preflight/checks.go

```diff
@@ -49,6 +55,40 @@ func (c HubKubeconfigCheck) Name() string {
 	return "HubKubeconfig check"
 }
 
+type DeployModeCheck struct {
+	Mode                  string
+	InternalEndpoint      bool
+	ManagedKubeconfigFile string
+}
+
+func (c DeployModeCheck) Check() (warningList []string, errorList []error) {
+	if c.Mode != InstallModeDefault && c.Mode != InstallModeHosted {
+		return nil, []error{errors.New("deploy mode should be default or hosted")}
+	}
+	if c.Mode == InstallModeDefault {
+		if c.ManagedKubeconfigFile != "" {
+			return nil, []error{errors.New("--managed-cluster-kubeconfig should not be set in default deploy mode")}
```

Comment:
I am not sure if we do need this check.

Reply:
I think we should not set the --managed-cluster-kubeconfig with default mode. Can it be set in default mode? @zhujian7

Reply:
I was thinking just ignoring it in default mode, but reporting an error also makes sense.